api/rds.proto

```diff
@@ -47,6 +47,10 @@ message WeightedCluster {
     // an upstream cluster is determined by its weight. The sum of weights
     // across all entries in the clusters array must add up to 100.
     google.protobuf.UInt32Value weight = 2;
+    // Optional endpoint metadata match criteria. Only endpoints in the upstream
+    // cluster with metadata matching that set in metadata_match will be
+    // considered. The filter name should be specified as "envoy.lb".
+    Metadata metadata_match = 3;
```

Comment:
This does make sense, for all the reasons that we discussed in the other issues. IMO there is zero downside to adding it here. It's opaque, and is not required to be used at all (and in fact it's unlikely Istio would use it). But it provides flexibility for folks that want to do label/complex based selection directly in the load balancer.